srtcore/sync.cpp

```diff
@@ -382,6 +382,53 @@ srt::sync::Mutex* srt::sync::UniqueLock::mutex()
     return &m_Mutex;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+//
+// CCondVar section (based on pthreads)
+//
+////////////////////////////////////////////////////////////////////////////////
+#ifndef USE_STL_CHRONO
+
+srt::sync::CCondVar::CCondVar()
+{
+    const int res = pthread_cond_init(&m_cv, NULL);
+    if (res != 0)
+        throw std::runtime_error("pthread_cond_init failed");
+}
+
+srt::sync::CCondVar::~CCondVar()
+{
+    pthread_cond_destroy(&m_cv);
+}
+
+bool srt::sync::CCondVar::wait_for(UniqueLock& lock, const steady_clock::duration& rel_time)
+{
+    return (SyncEvent::wait_for(&m_cv, &lock.mutex()->ref(), rel_time) != ETIMEDOUT);
```

Comment:
I think it makes much more sense to make `SyncEvent::wait_for` expect `srt::sync::Mutex` argument - just to reconsider.

Reply:
UniqueLock is to align with the [C++11 cv](https://en.cppreference.com/w/cpp/thread/condition_variable/wait_for).
Although the class cannot be mapped directly co std::condition_variable at the moment due to the return of this `wait_for()` function.

Reply:
We'll make do. This can be later changed into the aliased or crafted type of `cv_result`.